.github/workflows/cli-ci.yml

```diff
@@ -0,0 +1,89 @@
+name: cli/ubi CI
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'cli/go.mod'
+      - 'cli/ubi.go'
+      - 'bin/ubi'
+      - 'spec/cli_spec.rb'
+  pull_request:
+    paths:
+      - 'cli/go.mod'
+      - 'cli/ubi.go'
+      - 'bin/ubi'
+      - 'spec/cli_spec.rb'
+
+jobs:
+  cli-ci:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubicloud, ubicloud-arm]
+    name: cli/ubi CI - ${{matrix.runs-on}}
+    runs-on: ${{matrix.runs-on}}
+
+    env:
+      DB_USER: clover
+      DB_PASSWORD: nonempty
+      DB_NAME: clover_test
+
+    services:
+      postgres:
+        image: postgres:15.4
+        env:
+          POSTGRES_USER: ${{ env.DB_USER }}
+          POSTGRES_PASSWORD: ${{ env.DB_PASSWORD }}
+          POSTGRES_DB: ${{ env.DB_NAME }}
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+    - name: Perform superuser-only actions, then remove superuser
+      run: |
+        psql "postgres://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:5432/${{ env.DB_NAME }}" \
+          -c "CREATE EXTENSION citext; CREATE EXTENSION btree_gist; CREATE ROLE clover_password PASSWORD '${{ env.DB_PASSWORD }}' LOGIN; ALTER ROLE ${{ env.DB_USER }} NOSUPERUSER"
+
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Install ruby for ARM runners if not cached
+      if: matrix.runs-on == 'ubicloud-arm'
+      run: |
+        RUBY_VERSION="$(grep -E '^ruby ' .tool-versions | cut -d' ' -f2)"
+        if [ ! -f $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64.complete ]; then
+          git clone https://github.com/rbenv/ruby-build.git
+          sudo ./ruby-build/install.sh
+          sudo ruby-build "$RUBY_VERSION" $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64
+          sudo chown -R runner:runner $RUNNER_TOOL_CACHE/Ruby
+          touch $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64.complete
+          rm -rf ruby-build
+        fi
+
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: .tool-versions
+        bundler-cache: true
+
+    - name: Apply migrations
+      env:
+        CLOVER_DATABASE_URL: postgres://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:5432/${{ env.DB_NAME }}
+      run: bundle exec rake _test_up
+
+    - name: "Run go fmt and check it makes no changes"
```

Comment:
In addition to go fmt, you might consider adding some more linting (ie akin to rubocop or something). We use this: https://github.com/golangci/golangci-lint

Reply:
I'll look into this.

Reply:
Added a Rake task to do this locally, added it to cli-ci.yml, and fixed all identified issues.